assets/js/app/LocusZoom.js

```diff
@@ -15,35 +15,48 @@
     // Create a new instance by instance class and attach it to a div by ID
     // NOTE: if no InstanceClass is passed then the instance will use the Intance base class.
     //       The DefaultInstance class must be passed explicitly just as any other class that extends Instance.
-    exports.addInstanceToDivById = function(InstanceClass, id){
+    exports.addInstanceToDivById = function(id, datasource, layout, state){
```

Comment:
I wasn't really try to make any architectural statement; I was just trying to fix the broken build! See [this line](https://github.com/statgen/locuszoom/pull/7/files#diff-29448d642d7578e29c0da8c8b0382e91R62) in the test suite.

In order for that test to work that method needs to be exposed. Now tests should always evolve as the architecture evolves and not the other way around, so if it's "correct" to obscure that method then this test needs to change. Either way the build has to pass in order to merge.

As to the discussion of "what to export" I hadn't made that leap and I don't think we need to be particularly concerned with it right now either way. The only reason I refactored the base `LocusZoom.js` into a JS module with the exports syntax is so that mocha could pull it in as a module and we could have some automated tests. It worked just fine before without being a module at all. Now that it's built as a module we *can* get specific about what we expose, but I'd argue it's more of a distraction at this point. Just exposing every method is way simpler (yay KISS principle) and makes it easier to build unit tests for every method we build. Pulling it from the documentation and favoring other methods (as you've done in this branch) is a far more effective approach for our tiny user base, and if we actually have users digging through the source and plugging in undocumented functions in ways we don't expect them to then that's not our problem.